independent-projects/resteasy-reactive/server/jackson/src/main/java/org/jboss/resteasy/reactive/server/jackson/JacksonBasicMessageBodyReader.java

```diff
@@ -18,28 +23,65 @@
 
 public class JacksonBasicMessageBodyReader extends AbstractJsonMessageBodyReader {
 
-    protected final ObjectReader defaultReader;
+    private final ObjectMapper originalMapper;
+    private final ConcurrentMap<ObjectMapper, ObjectReader> contextResolverMap = new ConcurrentHashMap<>();
```

Comment:
Since `getProviders()` will return null when using the JacksonBasicMessageBodyReader class, there is no performance penalty here. 
Though, for the same reason, because `getProviders()` returns null, the only reason to have the getEffectiveReader or getEffectiveObjectMapper logics here is to reuse this code for the client and the server, so what about moving these methods into helpers that are used by the client and server implementations? wdyt?

Reply:
I understand that, but I still want `ServerJacksonMessageBodyReader` to do absolutely nothing so to speak and move all the complexity to `FullyFeaturedServerJacksonMessageBodyWriter` which only gets enabled under certain circumstances.

Reply:
It seems like `FullyFeaturedServerJacksonMessageBodyWriter` already has some kind of support for this...

Reply:
And it was added to support https://github.com/quarkusio/quarkus/issues/26152

Reply:
> I understand that, but I still want `ServerJacksonMessageBodyReader` to do absolutely nothing so to speak and move all the complexity to `FullyFeaturedServerJacksonMessageBodyWriter` which only gets enabled under certain circumstances.

Yes, and indeed this is the problem. ServerJacksonMessageBodyReader was used and hence the custom object mapper is not picked up.

Reply:
Ah sorry, this is about the reader...

In that case, I would very much like to follow a similar approach in the reader as we have in the writer - only add complexity for the server if it's **absolutely** necessary.
I don't want anything to slow down the most common path of the server part.

Reply:
so, are you ok with moving the getEffectiveReader and getEffectiveObjectMapper methods into a common helper, so both readers in the client and the server can use it?

Reply:
What would that look like?

Reply:
Finally, I scratched the idea of creating a common helper to reuse the getObjectMapperFromContext by the client and the server readers, because the code to share was very little, but let me know what you think about it.
I kept the class JacksonBasicMessageBodyReader with no changes as you suggested.
PR updated.

Reply:
I think that I have not properly explained what I mean (because I was confused about the reader / writer). 

What is going on now with your latest changes in the PR is that `getEffectiveReader` will go look up `Providers` (which will never be `null`), then look up the `ObjectMapper` via `providers.getContextResolver`.
This is the exact thing I want to avoid in the general case: There is absolutely no reason to go through this dance if there is no `ContextResolver<ObjectMapper>` defined (we know this at build time).
Therefore, what I want is something similar to what we have for the writer case: a server implementation of `JacksonBasicMessageBodyReader` which has no knowledge of this stuff and a new one like the one you added that does. The activation logic would be supplied at build time in a similar way we do for the writer.

Does this make sense?

P.S. I do not really worry about the client case as that can't really be a source of performance issues. I am however very worried about introducing additional overhead on the server part, hence my request for changes.

Reply:
I see now what you mean, thanks for clarifying! Will update the PR asap.

Reply:
Thanks! Sorry for not being clear earlier.

Reply:
PR updated!